tests/lib/bin/TODO

```diff
@@ -0,0 +1,115 @@
+themselves a full command to execute should usually end in '-do'.
+
+The commands might take one or multiple subcommands themselves if
+meaningful, subcommands should also use '-' as internal word
+separator.
+
+If a subcommand executes other commands it should usually be called
+"exec".
+
+All the commands invoked without arguments should either do something
+informative/non-destructive or print usage.
+
+All the commands should support at least a -h help option.
+
+To invoke a command here tests can do "$TESTSTOOLS"/cmd.
+
+Commands that are used very often or that are particularly useful to
+debug at a spread -debug shell prompt can be promoted to be on the
+PATH during tests, if an acceptable name can be found for them.
+
+This is done by putting symlinks to here from tests/bin which itself
+will be on the PATH.
+
+Commands expected to be on the PATH should follow more specific naming
+conventions, the name should either hint at testing or trade that off
+for brevity, but only when appropriate:
+
+* really frequently used commands can have short/pithy names (not, retry...)
+
+* commands that manage or deal with cross-test/cross-task state should be
+  prefixed with "tests." (tests.session)
+
+* otherwise the command should take some form of suffix like ".query", ".check"
+  that hints at observing, checking the system
+
+In general if none of these apply they should continue to be invoked
+via $TESTTOOLS, or possibly have a tests/lib/*.sh wrapping them if
+appropriate.
+
+TODO:
+
+rename tests/lib/bin to tests/lib/tools, define TESTSTOOLS in
+spread.yaml pointing at it
+
+create tests/bin, have ultimately this on the PATH and not tools
+
+symlinks and renames (plus fixes):
+
+tests/bin/MATCH -> tests/lib/tools/MATCH
+tests/bin/REBOOT -> tests/lib/tools/REBOOT
+tests/bin/not -> tests/lib/tools/not
+tests/bin/retry -> tests/lib/tools/retry (was retry-tool)
+
+tests/bin/tests.session -> tests/lib/tools/tests.session (was session-tool)
+  fix to have -h
+  convert --ACTION into subcommands
+  require "exec" subcommand to execute a command
+
+tests/lib/tools/memory-observe-do (was memory-tool)
+tests/lib/tools/version-compare (was version-tool)
+
+tests/bin/mountinfo.query -> tests/lib/tools/mountinfo.query (was mountinfo-tool)
+  (we might not do this and keep just tests/lib/tools/mountinfo and call it the long way)
+
+tests/bin/snapd.tool -> tests/lib/tools/snapd.tool (was snap-tool)
+  fix no args
+  fix to have -h
+  (we might not do this, this would be a special case, the name has
+   snapd in itself, it makes sense in /bin mostly for debugging,
+   otherwise it could stay just in tools as in itself it is not called
+   that often by tests, it would just become snapd-tool)
+
+We are then left with a mixture of commands with maybe one or a couple
+subcommands dealing with system state:
+
+* simple option rename them with a -state suffix in most cases:
+
+user-tool -> test/lib/tools/user-state
```

Comment:
How about "tests.user-state", "tests.apt-state", "tests.lxd-state", per one of the general ideas described above (commands that manage or deal with cross-test/cross-task state should be prefixed with "tests." (tests.session))?

Reply:
there is not immediate plan to put them on paths so they don't need a prefix. you'll have to call them like:

"$TESTSTOOLS"/apt-state

(they are used very little). We we want them on paths without the trampoline idea, then they would probably just be called tets.apt etc mabye.